CMakeLists.txt

```diff
@@ -59,6 +59,32 @@ if(CHECK_PIE_SUPPORTED)
   endif()
 endif()
 
+include(CheckCSourceCompiles)
+
+check_c_source_compiles("
```

Comment:
Thanks, this looks good.
Would you mind making one minor change, wrapping this in something like `if (CMAKE_CXX_COMPILER_ARCHITECTURE_ID MATCHES "RISCV32|RISCV64|RISCV128" OR
       CMAKE_SYSTEM_PROCESSOR MATCHES "riscv32|riscv64|riscv128"))` so we do not get extra output on non-RISCV platforms?

Reply:
@jan-wassenberg thanks for feedback, I have added the suggestions in the updated PR

Reply:
Thanks for updating!